examples/dmd/local_tw_csv.cpp

```diff
@@ -0,0 +1,640 @@
+    {
+        save_csv = true;
+    }
+
+    string outputPath = "run";
+    if (string(basename) != "") {
+        outputPath += "/" + string(basename);
+    }
+
+    if (mpi.Root()) {
+        const char path_delim = '/';
+        string::size_type pos = 0;
+        do {
+            pos = outputPath.find(path_delim, pos+1);
+            string subdir = outputPath.substr(0, pos);
+            mkdir(subdir.c_str(), 0777);
+        }
+        while (pos != string::npos);
+    }
+
+    CAROM::CSVDatabase csv_db;
+
+    string variable = string(var_name);
+    int nelements = 0;
+    csv_db.getIntegerArray(string(data_dir) + "/dim.csv", &nelements, 1);
+    CAROM_VERIFY(nelements > 0);
+    if (myid == 0)
+    {
+        cout << "Variable " << var_name << " has dimension " << nelements << "." << endl;
+    }
+
+    int dim = nelements;
+    vector<int> idx_state;
+    csv_db.getIntegerVector(string(data_dir) + "/" + string(spatial_idx_list) + ".csv", idx_state, false);
+    if (idx_state.size() > 0)
+    {
+        dim = idx_state.size();
+        if (myid == 0)
+        {
+            cout << "Restricting on " << dim << " entries out of " << nelements << "." << endl;
+        }
+    }
+
+    vector<double> indicator_val;
+    if (!train || numWindows > 0)
+    {
+        csv_db.getDoubleVector(string(outputPath) + "/indicator_val.csv", indicator_val, false);
+        if (indicator_val.size() > 0)
+        {
+            if (numWindows > 0)
+            {
+                CAROM_VERIFY(numWindows == indicator_val.size());
+            }
+            else
+            {
+                numWindows = indicator_val.size();
+            }
+            if (myid == 0)
+            {
+                cout << "Read indicator range partition with " << numWindows << " windows." << endl;
+            }
+        }
+    }
+
+    int npar = 0;
+    int num_train_snap;
+    vector<string> training_par_list;
+    if (train)
+    {
+        npar = csv_db.getLineCount(string(list_dir) + "/" + train_list + ".csv");
+        CAROM_VERIFY(npar == 1);
+
+        csv_db.getStringVector(string(list_dir) + "/" + train_list + ".csv", training_par_list, false);
+        stringstream par_ss(training_par_list[0]); // training DATASET
+        string par_dir;
+        getline(par_ss, par_dir, ',');
+
+        vector<int> snap_bound;
+        csv_db.getIntegerVector(string(data_dir) + "/" + par_dir + "/" + temporal_idx_list + ".csv", snap_bound, false);
+        if (snap_bound.size() > 0)
+        {
+            snap_bound[0] -= 1;
+            snap_bound[1] -= 1;
+            CAROM_VERIFY(snap_bound.size() == 2);
+            num_train_snap = snap_bound[1] - snap_bound[0] + 1;
+        }
+        else
+        {
+            num_train_snap = csv_db.getLineCount(string(list_dir) + "/" + par_dir + ".csv");
+        }
+
+        CAROM_VERIFY(windowOverlapSamples < windowNumSamples);
+        numWindows = (windowNumSamples < infty) ? ceil(num_train_snap / windowNumSamples) : 1;
```

Comment:
I think this should be `ceil((double) num_train_snap / (double) windowNumSamples)`
Note that the output of 
cout << "ceil " << ceil(1.5) << ", " << ceil(3/2) << ", " << ceil((double) 3 / (double) 2) << endl;
is
ceil 2, 1, 2